public/login.php

```diff
@@ -177,19 +193,16 @@
 
     if (isset($oldLanguagePreferred)) {
         $parameters['language'] = $oldLanguagePreferred;
-    } else {
-        if (is_string($_GET['language'])) {
-            $parameters['language'] = $_GET['language'];
-        }
+    } elseif (is_string($_GET['language'])) {
+        $parameters['language'] = $_GET['language'];
     }
 }
 
 if (isset($serviceUrl)) {
     $defaultTicketName = isset($_GET['service']) ? 'ticket' : 'SAMLart';
     $ticketName = $casconfig->getOptionalValue('ticketName', $defaultTicketName);
 
-    $attributeExtractor = new AttributeExtractor();
-    $mappedAttributes = $attributeExtractor->extractUserAndAttributes($as->getAttributes(), $casconfig);
+        $mappedAttributes = $attributeExtractor->extractUserAndAttributes($as->getAttributes());
```

Comment:
I think we should pass in the current `$state` (from `$as->getAuthDataArray()`) through to the processing chain. This is because the authentication will set certain `$state` variables that some authprocs rely on being present.

Reply:
Done